tests/test_abcs_parameters.py

```diff
@@ -0,0 +1,188 @@
+    annotation_path: str,
+    # set videos
+    video_uris: str,
+    # set brand
+    brand_name: str,
+    brand_variations: list[str],
+    branded_products: list[str],
+    branded_products_categories: list[str],
+    branded_call_to_actions: list[str],
+    # set thresholds
+    early_time_seconds: float,
+    confidence_threshold: float,
+    face_surface_threshold: float,
+    logo_size_threshold: float,
+    avg_shot_duration_seconds: float,
+    dynamic_cutoff_ms: float,
+    # set model
+    llm_name: str,
+    video_size_limit_mb: int,
+    max_output_tokens: int,
+    temperature: float,
+    top_p: float,
+    top_k: float):
+
+    self.project_id = project_id
+    self.project_zone = project_zone
+    self.bucket_name = bucket_name
+    self.knowledge_graph_api_key = knowledge_graph_api_key
+    self.bigquery_dataset = bigquery_dataset
+    self.bigquery_table = bigquery_table
+    self.assessment_file = assessment_file
+    self.use_annotations = use_annotations
+    self.use_llms = use_llms
+    self.verbose = verbose
+    self.annotation_path = annotation_path
+    # set videos
+    self.video_uris = video_uris
+    # set brand
+    self.brand_name = brand_name
+    self.brand_variations = brand_variations
+    self.branded_products = branded_products
+    self.branded_products_categories = branded_products_categories
+    self.branded_call_to_actions = branded_call_to_actions
+    # set thresholds
+    self.early_time_seconds = early_time_seconds
+    self.confidence_threshold = confidence_threshold
+    self.face_surface_threshold = face_surface_threshold
+    self.logo_size_threshold = logo_size_threshold
+    self.avg_shot_duration_seconds = avg_shot_duration_seconds
+    self.dynamic_cutoff_ms = dynamic_cutoff_ms
+    # set model
+    self.llm_name = llm_name
+    self.video_size_limit_mb = video_size_limit_mb
+    self.max_output_tokens = max_output_tokens
+    self.temperature = temperature
+    self.top_p = top_p
+    self.top_k = top_k
+
+
+def test_not_empty_abcd_params():
+  """Tests that all brand parameters are provided"""
+
+  args = ArgsMock(project_id="tighlock-test", project_zone="us-central1",
+    bucket_name="abcd-detector-input", knowledge_graph_api_key="asdfds", bigquery_dataset="abcd_detector_ds", bigquery_table="my_table",
+    assessment_file="", use_annotations=True, use_llms=True, verbose=True, annotation_path="",video_uris="gs://abcd-detector-input/Google/videos/",
+    brand_name="Google", brand_variations="Google,google", branded_products="Google pixel, Google pixel buds, Google pixel watch",
+    branded_products_categories="phone, watch, buds", branded_call_to_actions="buy it!, buy", early_time_seconds=5, confidence_threshold=0.5,
+    face_surface_threshold=0.5, logo_size_threshold=0.5, avg_shot_duration_seconds=3, dynamic_cutoff_ms=3000, llm_name="asd",
+    video_size_limit_mb=50, max_output_tokens=8000, temperature=1, top_k=0.1, top_p=0.1
+  )
+
+  config = build_abcd_params_config(args)
```

Comment:
Move to method of class.

Reply:
For example, mock sys.argv

```
import argparse
from unittest import TestCase, mock

def main():
    parser = argparse.ArgumentParser()
    parser.add_argument('--name', required=True)
    args = parser.parse_args()
    print(f'Hello {[args.name](http://args.name/)}')

class TestArgparse(TestCase):
    def test_main(self):
        with mock.patch('sys.argv', ['my_script.py', '--name', 'John']):
            config = Configuration()
           config.from_args()
          # assert values
```

Reply:
Would recommend keeping as-is to follow [best practices](https://engdoc.corp.google.com/eng/doc/devguide/py/style/index.md?cl=head#:~:text=Never%20use%20staticmethod%20unless%20forced%20to%20in%20order%20to%20integrate%20with%20an%20API%20defined%20in%20an%20existing%20library.%20Write%20a%20module%2Dlevel%20function%20instead)

Reply:
+1 we need those functions separate from the Config class to be able to test how this object is being built.

Reply:
That may be the wrong best practice, there is no static method being suggested here.
On the contrary it would have access to self to set the parameters.

Reply:
This is the best practices I was referring to, so maybe we can consolidate to this conversation.

This is written as a builder function that creates the configuration class and populates it in one go. For it to have the same function inside of a class, it would have to be a `@classmethod`.

Reply:
Nothing prevents it from being an instance method, as a technical call.  Ana's call as an implementation call.

Reply:
I prefer the functional approach because it's self documenting, while a class that needs to have a method called after instantiation is less obvious to me and feels very error-prone. I'd prefer a `@classmethod` to a regular method. I'd probably recommend a regular method get refactored to a function or `@classmethod`, but agree that it's the code owner's choice so long as there is a consistency in style.